Implement "Introduce named variable for captured block" rendering strategy

msalzburg opened the pull request:
This PR implements the "Introduce named variable for captured block" strategy from Issue #32. 

It defines a #block and a #block? method to access the captured, nested block content.

jensljungblad commented:
@msalzburg Looking at https://github.com/github/actionview-component, it seems like they went with this strategy as well, but named the variable `content`. Should we do the same perhaps?

jensljungblad commented:
We could also add a test case for validating its presence.

msalzburg commented:
Hi @jensljungblad - I updated this branch by merging your small refactoring  and renamed the block handling methods to .content and .content? (again). Since the content block is optional I am not sure which presence you would like to validate? 

Maybe we can squeeze this feature in your upcoming 1.0.0 release?

jensljungblad commented on README.md:
```diff
@@ -269,23 +269,27 @@ end
 Elements can be thought of as isolated subcomponents, and they are defined on the component. Passing `multiple: true` makes it a repeating element, and passing a block lets us declare attributes on our elements, in the same way we declare attributes on components.
 
-In order to populate them with data, we pass a block to the component helper, which yields the component, which lets us set attributes and blocks on the element in the same way we do for components:
+In order to populate them with data, we pass a content block to the component helper, which yields the component, which lets us set attributes and content blocks on the element in the same way we do for components:
```
```suggestion
In order to populate them with data, we pass a block to the component helper, which yields the component, which lets us set attributes and content blocks on the element in the same way we do for components:
```

jensljungblad commented on README.md:
```diff
@@ -122,13 +122,13 @@ end
 
 <div class="alert alert--<%= alert.context %>" role="alert">
-  <%= alert %>
+  <%= alert.content %>  
```
```suggestion
  <%= alert.content %>
```

jensljungblad commented:
I'll give you a chance to reply to my comment above but I'm happy to merge as is :)

jensljungblad commented:
Eh, decided to just merge it :) Maybe I'll have a go at adding the tests.

jensljungblad commented:
Thanks a lot for the contribution!

jensljungblad commented:
@msalzburg Released as `1.0.0.rc2`.

msalzburg commented:
Awesome - thanks!
